### PR TITLE
Create endpoint to list available tokenIDs

### LINKED ISSUE #1256: Create endpoint to list available tokenIDs
### Description

Create the following endpoint.

```
HTTP GET /api/v3/token/available-ids
RPC  get.token.available-ids
```

### Acceptance Criteria

- The endpoints listed above are created
- The response is matching as described below
- Swagger is up-to-date
- New integration tests are added

### Additional information
*Request params:*
| Parameter | Type      | Validation | Default   | Comment     |
| -------------- | ----------- | ------------- | ----------- | ---------------- |
| limit            | Number | [0, 100]    | 10          | -                   |
| offset         | Number | [0, ∞)        | 0            | -                   |

*Responses:*

200 OK
```jsonc
{
  "data": {
    "tokenIDs": [
      "04000000000000000000",
    ],
  },
  "meta": {
    "limit": 10,
    "offset": 0,
    "total": 1
  }
}
```

400 Bad Request
```jsonc
{
  "error": true,
  "message": "Unknown input parameter(s): <param_name>"
}
```

### PR DESCRIPTION
**Pull request by vardan10**
### What was the problem?
This PR resolves #1256

### How was it solved?

- [x] Endpoints listed above are created
- [x] Add hardcoded response
- [x] Swagger is up-to-date
- [x] Integration tests are added
- [x] Update docs

### How was it tested?

- [x] Local

### CODE REVIEW

File: tests/integration/api_v3/http/tokenIDs.test.js
Comment (priojeetpriyom) [inline tests/integration/api_v3/http/tokenIDs.test.js:1]: I guess we will increase the test coverage as that point is not marked as completed yet. But still mentioning for completeness. Could we please add some tests covering limit, offset and bad input scenarios? Also, we can validate that the number of tokenIds returns falls within some range. Ex: 

https://github.com/LiskHQ/lisk-service/blob/feature/1157-prepare-lisk-service-v070-for-beta/tests/integration/api_v3/http/posStakers.test.js#L125-L171
https://github.com/LiskHQ/lisk-service/blob/feature/1157-prepare-lisk-service-v070-for-beta/tests/integration/api_v3/http/posStakers.test.js#L125-L171

File: tests/schemas/api_v3/tokenIDs.schema.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import Joi from 'joi';
+
+const tokenIDsMetaSchema = {
+	count: Joi.number().integer().min(0).required(),
+	offset: Joi.number().integer().min(0).required(),
+	total: Joi.number().integer().min(0).required(),
+};
+
+const tokenIDsSchema = {
+	tokenIDs: Joi.array().required(),
```
Comment (priojeetpriyom) [inline tests/schemas/api_v3/tokenIDs.schema.js:26]: We can add tokenID regex validation for the items

File: tests/integration/api_v3/rpc/tokenIDs.test.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+describe('get.token.ids', () => {
+	beforeAll(async () => {
+	});
+
+	it('returns tokens info when call with address', async () => {
+		const response = await getTokensIds({});
+		expect(response).toMap(jsonRpcEnvelopeSchema);
+		const { result } = response;
+		expect(result.data).toMap(tokenIDsSchema);
+		expect(result.meta).toMap(tokenIDsMetaSchema);
```
Comment (priojeetpriyom) [inline tests/integration/api_v3/rpc/tokenIDs.test.js:43]: Checking the `goodRequestSchemaFortokenIDs` would already validate both.

```suggestion
		expect(result.data).toMap(goodRequestSchemaFortokenIDs);
```

File: tests/schemas/api_v3/tokenIDs.schema.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+};
+
+const goodRequestSchemaFortokenIDs = {
```
Comment (priojeetpriyom) [inline tests/schemas/api_v3/tokenIDs.schema.js:29]: Previously, the container schema used to be named `goodRequestSchema`, then we started naming in respect to the api name (ex: tokenIDsResponseSchema in this case). Maybe we can do either of below or something similar

```suggestion
const goodResponseSchemaFortokenIDs = {
```
OR

```suggestion
const tokenIDsResponseSchema = {
```

File: services/blockchain-indexer/methods/dataService/controllers/token.js [hunk 4]
```diff
@@ -75,9 +75,24 @@ const getTokenConstants = async () => {
 	return constants;
 };
 
+// TODO: fetch available tokens after indexing
```
Comment (vardan10) [inline services/blockchain-indexer/methods/dataService/controllers/token.js:78]: To be added after https://github.com/LiskHQ/lisk-service/issues/1383

File: tests/integration/api_v3/http/tokenIDs.test.js [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+	goodResponseSchemaFortokenIDs,
+} = require('../../../schemas/api_v3/tokenIDs.schema');
+
+const baseUrl = config.SERVICE_ENDPOINT;
+const baseUrlV3 = `${baseUrl}/api/v3`;
+const endpoint = `${baseUrlV3}/token/ids`;
+
+describe('Token IDs API', () => {
+	it('Should retrieves token info when called without any params', async () => {
+		const response = await api.get(`${endpoint}`);
```
Comment (nagdahimanshu) [inline tests/integration/api_v3/http/tokenIDs.test.js:34]:
```suggestion
		const response = await api.get(endpoint);
```

File: tests/integration/api_v3/rpc/tokenIDs.test.js [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+describe('get.token.ids', () => {
+	it('returns tokens info when call with address', async () => {
+		const response = await getTokensIds({});
+		expect(response).toMap(jsonRpcEnvelopeSchema);
+		const { result } = response;
+		expect(result).toMap(goodResponseSchemaFortokenIDs);
+		expect(result.data.tokenIDs.length).toBeLessThanOrEqual(10);
+	});
+
+	it('Returns list of stakers when requested with known validator name and offset=1', async () => {
```
Comment (nagdahimanshu) [inline tests/integration/api_v3/rpc/tokenIDs.test.js:43]:
```suggestion
	it('Should retrieves token info when called with offset=1', async () => {
```

File: tests/integration/api_v3/rpc/tokenIDs.test.js [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+
+	it('Returns list of stakers when requested with known validator name and limit=5', async () => {
+		const response = await getTokensIds({ limit: 5 });
+		expect(response).toMap(jsonRpcEnvelopeSchema);
+		const { result } = response;
+		expect(result).toMap(goodResponseSchemaFortokenIDs);
+		expect(result.data.tokenIDs.length).toBeLessThanOrEqual(5);
+	});
+
+	it('Returns list of stakers when requested with known validator name, offset=1 and limit=5', async () => {
```
Comment (nagdahimanshu) [inline tests/integration/api_v3/rpc/tokenIDs.test.js:59]:
```suggestion
	it('Should retrieves token info when called with offset=1 and limit=5', async () => {
```

File: tests/integration/api_v3/rpc/tokenIDs.test.js [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+	invalidParamsSchema,
+	jsonRpcEnvelopeSchema,
+} = require('../../../schemas/rpcGenerics.schema');
+
+const {
+	goodResponseSchemaFortokenIDs,
+} = require('../../../schemas/api_v3/tokenIDs.schema');
+
+const wsRpcUrl = `${config.SERVICE_ENDPOINT}/rpc-v3`;
+const getTokensIds = async (params) => request(wsRpcUrl, 'get.token.ids', params);
```
Comment (nagdahimanshu) [inline tests/integration/api_v3/rpc/tokenIDs.test.js:32]:
```suggestion
const getTokensIDs = async (params) => request(wsRpcUrl, 'get.token.ids', params);
```

File: tests/schemas/api_v3/tokenIDs.schema.js [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+	total: Joi.number().integer().min(0).required(),
+};
+
+const tokenIDsSchema = {
+	tokenIDs: Joi.array().items(Joi.string().pattern(regex.TOKEN_ID)).required(),
+};
+
+const goodResponseSchemaFortokenIDs = {
+	data: Joi.object(tokenIDsSchema).required(),
+	meta: Joi.object().required(),
```
Comment (nagdahimanshu) [inline tests/schemas/api_v3/tokenIDs.schema.js:33]:
```suggestion
	meta: Joi.object(tokenIDsMetaSchema).required(),
```

File: tests/schemas/api_v3/tokenIDs.schema.js [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+};
+
+module.exports = {
+	tokenIDsSchema: Joi.object(tokenIDsSchema).required(),
+	tokenIDsMetaSchema: Joi.object(tokenIDsMetaSchema).required(),
```
Comment (nagdahimanshu) [inline tests/schemas/api_v3/tokenIDs.schema.js:38]: We are exporting `tokenIDsMetaSchema` but it's not being used anywhere. We can remove it from export.

File: services/gateway/sources/version3/tokenIDs.js [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+module.exports = {
+	type: 'moleculer',
+	method: 'indexer.token.ids',
+	params: {
+		offset: '=,number',
+		limit: '=,number',
+	},
+	definition: {
+		data: {
+			tokenIDs: '=',
```
[the remark this answers is not in the document]
Comment (sameersubudhi) [reply services/gateway/sources/version3/tokenIDs.js:26]: It is fine as is. No changes necessary.

File: tests/schemas/api_v3/tokenIDs.schema.js [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+const regex = require('./regex');
+
+const tokenIDsMetaSchema = {
+	count: Joi.number().integer().min(0).required(),
+	offset: Joi.number().integer().min(0).required(),
+	total: Joi.number().integer().min(0).required(),
+};
+
+const tokenIDsSchema = {
+	tokenIDs: Joi.array().items(Joi.string().pattern(regex.TOKEN_ID)).required(),
```
Comment (sameersubudhi) [inline tests/schemas/api_v3/tokenIDs.schema.js:28]:
```suggestion
	tokenIDs: Joi.array().items(Joi.string().pattern(regex.TOKEN_ID)).min(1).required(),
```

File: tests/integration/api_v3/rpc/tokenIDs.test.js [hunk 13]
```diff
@@ -0,0 +1,81 @@
...
+		expect(result).toMap(goodResponseSchemaFortokenIDs);
+		expect(result.data.tokenIDs.length).toBeLessThanOrEqual(10);
+	});
+
+	it('Should retrieves token ids when called with offset=1', async () => {
+		const response = await getTokensIDs({ offset: 1 });
+		expect(response).toMap(jsonRpcEnvelopeSchema);
+		const { result } = response;
+		expect(result).toMap(goodResponseSchemaFortokenIDs);
+		expect(result.data.tokenIDs.length).toBeLessThanOrEqual(10);
```
Comment (sameersubudhi) [inline tests/integration/api_v3/rpc/tokenIDs.test.js:48]:
```suggestion
		expect(result.data.tokenIDs.length).toBeGreaterThanOrEqual(0);
		expect(result.data.tokenIDs.length).toBeLessThanOrEqual(10);
```

File: tests/integration/api_v3/rpc/tokenIDs.test.js [hunk 14]
```diff
@@ -0,0 +1,81 @@
...
+		expect(result).toMap(goodResponseSchemaFortokenIDs);
+		expect(result.data.tokenIDs.length).toBeLessThanOrEqual(5);
+	});
+
+	it('Should retrieves token ids when called with offset=1 and limit=5', async () => {
+		const response = await getTokensIDs({ offset: 1, limit: 5 });
+		expect(response).toMap(jsonRpcEnvelopeSchema);
+		const { result } = response;
+		expect(result).toMap(goodResponseSchemaFortokenIDs);
+		expect(result.data.tokenIDs.length).toBeLessThanOrEqual(5);
```
Comment (sameersubudhi) [inline tests/integration/api_v3/rpc/tokenIDs.test.js:64]:
```suggestion
		expect(result.data.tokenIDs.length).toBeGreaterThanOrEqual(0);
		expect(result.data.tokenIDs.length).toBeLessThanOrEqual(5);
```

File: tests/integration/api_v3/http/tokenIDs.test.js [hunk 15]
```diff
@@ -18,22 +18,58 @@ const { api } = require('../../../helpers/api');
...
+	});
+
+	it('Should retrieves token ids when called with offset=1 and limit=5', async () => {
+		const response = await api.get(`${endpoint}?offset=1&limit=5`);
+		expect(response).toMap(goodRequestSchema);
+		expect(response).toMap(goodResponseSchemaFortokenIDs);
+		expect(response.data.tokenIDs.length).toBeLessThanOrEqual(5);
+	});
+
+	it('Invalid limit -> bad request', async () => {
```
Comment (priojeetpriyom) [inline tests/integration/api_v3/http/tokenIDs.test.js:65]: We can refactor other tests like below

```suggestion
	it('should return bad request when called with Invalid limit', async () => {
```

File: tests/integration/api_v3/http/tokenAvailableIDs.test.js [hunk 16]
```diff
@@ -0,0 +1,74 @@
...
+	});
+
+	it('should retrieves token ids when called with limit=5', async () => {
+		const response = await api.get(`${endpoint}?limit=5`);
+		expect(response).toMap(goodResponseSchemaFortokenAvailableIDs);
+		expect(response.data.tokenIDs.length).toBeGreaterThanOrEqual(1);
+		expect(response.data.tokenIDs.length).toBeLessThanOrEqual(5);
+	});
+
+	it('should retrieves token ids when called with offset=1 and limit=5', async () => {
```
Comment (sameersubudhi) [inline tests/integration/api_v3/http/tokenAvailableIDs.test.js:53]:
```suggestion
	it('should retrieve available token ids when called without any params', async () => {
		const response = await api.get(endpoint);
		expect(response).toMap(goodResponseSchemaFortokenAvailableIDs);
		expect(response.data.tokenIDs.length).toBeGreaterThanOrEqual(1);
		expect(response.data.tokenIDs.length).toBeLessThanOrEqual(10);
	});

	it('should retrieve available token ids when called with offset=1', async () => {
		const response = await api.get(`${endpoint}?offset=1`);
		expect(response).toMap(goodResponseSchemaFortokenAvailableIDs);
		expect(response.data.tokenIDs.length).toBeGreaterThanOrEqual(0);
		expect(response.data.tokenIDs.length).toBeLessThanOrEqual(10);
	});

	it('should retrieve available token ids when called with limit=5', async () => {
		const response = await api.get(`${endpoint}?limit=5`);
		expect(response).toMap(goodResponseSchemaFortokenAvailableIDs);
		expect(response.data.tokenIDs.length).toBeGreaterThanOrEqual(1);
		expect(response.data.tokenIDs.length).toBeLessThanOrEqual(5);
	});

	it('should retrieve available token ids when called with offset=1 and limit=5', async () => {
```

**Review by sameersubudhi — APPROVED** [review]
Look good but please consider the following.

**Review by nagdahimanshu — APPROVED** [review]
LGTM 👍

File: docs/api/version3.md [hunk 17]
```diff
@@ -1145,6 +1145,55 @@ Get module summary from the Token module
...
+
+- HTTP GET `/api/v3/token/available-ids`
+- RPC `get.token.available-ids`
+
+#### Request parameters
+
+| Parameter | Type | Validation | Default | Comment |
+| --------- | ---- | ---------- | ------- | ------- |
+| limit | Number | `[1,100]` | 10 |  |
+| offset | Number | `[1,Inf)` | 0 |  |
```
Comment (priojeetpriyom) [inline docs/api/version3.md:1162]:
```suggestion
| offset | Number | `[0,Inf)` | 0 |  |
```

File: docs/api/version3.md [hunk 18]
```diff
@@ -1145,6 +1145,55 @@ Get module summary from the Token module
 https://service.lisk.com/api/v3/token/summary
 ```
 
+### Token Available IDs
```
Comment (sameersubudhi) [inline docs/api/version3.md:1148]: @vardan10 Thank you for updating the document. Could also please add an entry in the list above?

**Review by priojeetpriyom — APPROVED** [review]

File: docs/api/version3.md (hunk 17 above)
Comment (vardan10) [reply docs/api/version3.md]: Applied

File: docs/api/version3.md (hunk 18 above)
Comment (vardan10) [reply docs/api/version3.md:1149]: Applied

**Review by sameersubudhi — APPROVED** [review]
Thank you for all the changes! LGTM 👍

**Review by priojeetpriyom — APPROVED** [review]

**Review by nagdahimanshu — APPROVED** [review]

**Pull request merged**